### ISSUE TITLE
This is not an issue but useful

### ISSUE THREAD

**Issue by zw5** [KeyPuncher/WiinUPro#14]
I have a lot of knowledge in spanish so i can translate the app to spanish.

**Comment by StrScr** [comment]
I think localization is definitely something that every project needs at some point. I'll try to start globalizing the project in a fork of my own, and we'll see about submitting a PR later on. 😄

**Comment by insaneTurnip** [comment]
Hey there, I'm up if you need a French translation.

**Comment by KeyPuncher** [comment]
I'm working on a localization solution right now, which is why I assigned it to myself as a way to say _in progress._

**Comment by zw5** [comment]
I'm still here if you want me to translate the strings to spanish.

**Comment by KeyPuncher** [comment]
Nah, it's just taken me forever to get to it and since I didn't design it with multi-lingual support in mind it's more difficult to add it afterwards.

I setup a sheet for translations here [https://docs.google.com/spreadsheets/d/1GBMtDP-JgyXad1y14ACACvXAcUiwBWiiSj1K1h9dmSs/edit?usp=sharing](google spreadsheet)

If @zw5 could add Spanish that would be awesome.

@Wrexes if you're still available for French that would also be awesome.

**Comment by zw5** [comment]
I added spanish just now.

**Comment by zw5** [comment]
It would be really useful if you could provide me with a preview build or something that could allow me to see the language options in the application so I can check if everything "feels" right and there isn't any silly translation mistakes.

**Comment by Wrexes** [comment]
Alright, I'll see if I get time to do it in between my insomnia and stream schedule. 160 items should be okay.
Also I second @zw5's last message though I am now using Linux.
Do you think it could run through wine despite being a .NET program ? I just need the interface too.
I never launched en exe on a *NIX platform so I have no idea...

**Comment by KeyPuncher** [comment]
The code isn't ready yet but I'll hand it out once it is with the Dummy devices enabled so you can simulate a controller connection.

I'm not sure if it'll run with wine or not, I might get a chance to test that out tomorrow. I'm more concerned about the vJoy native library and bluetooth references not working.

**Comment by Wrexes** [comment]
Hey, I'm currently working on the French translation.
I wrote some comments here and there to avoid any misunderstanding which could result in obfuscation rather than clarification.

On the subject of a Linux port, maybe making another issue about it would be interesting.
Besides, I have found a [Linux virtual controller engine](https://github.com/jgeumlek/MoltenGamepad) (including a Wiimote driver) developed on Arch Linux 64bits (same as my system) as well as having a Debian package. It's still maintained, too !
I would need to find a new battery to test WiinUPro with my old WiiU Pro Controller because it doesn't  turn on anymore, but I have some AA cells and a Wiimote + Nunchuck, and even my old IR bar (and Wii to power it) to try things out.
Let me know if you'd like us to open a new issue and start testing things !

**Comment by Wrexes** [comment]
Hello,
I am done with the French translation.
I am sorry it took so long. With the current state of things in France (new lockdown due to COVID-19 resurgence) and my insomnia going wild, it's been some pretty rough weeks lately.
I hope you can understand.

**Comment by KeyPuncher** [comment]
No worries, this isn't a rush. I've been busy with school work and work has also been especially busy recently.
I still haven't finished the framework for translation yet. I appreciate you putting in the time.

**Comment by KeyPuncher** [comment]
Just pushed a lengthy change that I've been working on the past few months, when time permitted, that was necessary in order to support this. Only a few more changes to make for the implementation.

**Comment by KeyPuncher** [comment]
@zw5 @SirWrexes 

Finally finished the integration. Attached is a preview build.
Also there are two new fields in the google sheet:

- Settings_Minimize
- Settings_ProfileQ

Also the French translation for Calibration_IR_Screen is missing.
Thanks for the help!

[64-Bit.zip](https://github.com/KeyPuncher/WiinUPro/files/6291032/64-Bit.zip)

**Comment by zw5** [comment]
Hi. Looking at the builds, I can see that accents and special characters (like "ñ", "é", "ä") don't render properly. I think it's an issue with the font you're using. Also some buttons are way too small for the text that they're showing so you can only read a few letters of the whole word.
Examples:
![image](https://user-images.githubusercontent.com/27164429/114478323-96735180-9bc3-11eb-92eb-f00736246469.png)
![image](https://user-images.githubusercontent.com/27164429/114478340-a0955000-9bc3-11eb-91ac-b01ba40db017.png)

**Comment by KeyPuncher** [comment]
Oh, looks like it's the encoding of the file. Thankfully you can just drop this one over the old one and it should look different.

[lang.zip](https://github.com/KeyPuncher/WiinUPro/files/6301461/lang.zip)

Also looks like I missed some areas. I'll need to add new keys to the google doc. I'll evaluate those areas and try to add them as soon as I can.

**Comment by SirWrexes** [comment]
Hi !
I'll fill the missing fields when I can, my bad.
In terms of testing it, it'll have to wait a few months.
I'm saving up for a new PC, and in the meantime, I only have Linux.

**Comment by SkorX** [comment]
Hello,
I added full polish translation in your Google Sheet 😀

**Comment by KeyPuncher** [comment]
> Hello, I added full polish translation in your Google Sheet 😀

I saw, thank you for the translations!
Here's a debug build that will allow you to connect test devices to check out the translations. I already found a couple of areas where I will need to adjust the UI or at least add a tooltip with the full text.

[WiinUPro Debug.zip](https://github.com/KeyPuncher/WiinUPro/files/7606293/WiinUPro.Debug.zip)

**Comment by SkorX** [comment]
**I finally (after really long delay) fixed polish translations and it's ready now 😀**

**I also left some notes (which i copied here, so you don't have to look for it):**
- I noticed many places where you have hardcoded translations (it appeared in english, even when language is changed), so you may consider changing it.
- Also some buttons (and labels) seems to be artificially narrow (even when there are some space around) and because of it translations have to shortened (sadly polish translations can be even 15% longer than original english text (eg. "Start Minimized" should be translated to "Uruchom zminimalizowany" what is 53% longer (number of letters, but actual text length may depend on font which is used))).

**Comment by KeyPuncher** [comment]
> **I finally (after really long delay) fixed polish translations and it's ready now 😀**
> 

I appreciate it! Still speedy fast compared to my release schedule. I'll see about addressing these over the summer while I'm out of school. This semester's course is very time consuming and I've only got about 3 weeks left.

**Comment by SirWrexes** [comment]
Hi, sorry for the lack of activity. Lots of stuff happened since last time, including my laptop getting stolen so I don't think I'm buying a new PC soon.
I regret to say you should count me out for the French translation testing and maintaining. :/
